File: core/luana-core-extension-sdk/src/luana_core_extension_sdk/brand_context.py
"""BrandContext — per-tenant frozen context passed to extension handlers.

Per checkpoint §7.5.2 D3 — 9 fields FROZEN at v0.1.0. Future optional fields
opcionales agregables sin breaking bump. Handler that uses only a subset of
fields MUST continue working when new optional fields are appended.

NO PII fields — safe to log. tenant_id + tenant_profile_id are opaque UUIDs.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal
from uuid import UUID


@dataclass(frozen=True, slots=True, kw_only=True)
class BrandContext:
    """Per-tenant context passed to extension handlers (CC-4 namespace + brand routing).

    Per checkpoint §7.5.2 D3 — 9 fields FROZEN at v0.1.0. Future optional fields
    opcionales agregables sin breaking bump. Handler that uses only a subset of
    fields MUST continue working when new optional fields are appended.

    NO PII fields — safe to log. tenant_id + tenant_profile_id are opaque UUIDs.
    """

    tenant_id: UUID
    brand_slug: Literal["nicolify", "vitalia", "comunify", "lupulo", "test-brand"]
    plan_tier: str  # tier_id from EP-17 brand-registered
    locale: str  # ISO 639-1 + region (es-AR / es-MX / es-CL / en-US)
    feature_flags: dict[str, bool]  # tenant-level flag map (brand-injected)
    tenant_profile_id: UUID
    vertical_kind: Literal["marketing", "medical", "creator-economy", "gastronomy"]
    compliance_flags: dict[str, bool]  # e.g. {"hipaa_required": True} per Vitalia
    pii_policy: Literal["standard", "medical", "creator", "gastronomy"]

File: core/@luana/extension-sdk/src/brand-context.ts
/**
 * BrandContext — per-tenant context passed to extension handlers.
 *
 * TypeScript mirror of Python BrandContext frozen dataclass (9 fields).
 * Per §7.5.2 D3 — frozen at v0.1.0. Future optional fields may be added
 * without breaking bump.
 *
 * NO PII fields — safe to include in trace logs.
 */

export type BrandSlug = "nicolify" | "vitalia" | "comunify" | "lupulo" | "test-brand";

export type VerticalKind = "marketing" | "medical" | "creator-economy" | "gastronomy";

export type PiiPolicy = "standard" | "medical" | "creator" | "gastronomy";

export interface BrandContext {
  /** Opaque tenant UUID (snake_case: tenant_id) */
  tenantId: string;
  /** Brand slug from allowlist (snake_case: brand_slug) */
  brandSlug: BrandSlug;
  /** Plan tier identifier (snake_case: plan_tier) */
  planTier: string;
  /** ISO 639-1 + region locale, e.g. "es-AR" | "es-MX" | "en-US" (snake_case: locale) */
  locale: string;
  /** Tenant-level feature flag map (snake_case: feature_flags) */
  featureFlags: Record<string, boolean>;
  /** Opaque tenant profile UUID (snake_case: tenant_profile_id) */
  tenantProfileId: string;
  /** Vertical domain kind (snake_case: vertical_kind) */
  verticalKind: VerticalKind;
  /** Compliance flags, e.g. { hipaa_required: true } (snake_case: compliance_flags) */
  complianceFlags: Record<string, boolean>;
  /** PII handling policy (snake_case: pii_policy) */
  piiPolicy: PiiPolicy;
}
